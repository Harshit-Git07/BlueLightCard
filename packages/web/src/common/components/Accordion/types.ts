export interface AccordionProps {
  title: string;
  children: React.ReactNode;
  onClickOpen?: () => void;
}
