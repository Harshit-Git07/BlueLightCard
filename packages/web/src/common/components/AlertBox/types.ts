export interface AlertBoxProps {
  alertType: string;
  title: string;
  description: any;
}

export interface style {
  textColor: string;
  backgroundColor: string;
}

export interface config {
  [key: string]: style;
}
