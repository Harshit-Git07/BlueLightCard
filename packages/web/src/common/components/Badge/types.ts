export type BgColorString = `bg-${string}`;

export type BadgeProps = {
  label: string;
  color: BgColorString;
  size: 'small' | 'large';
};
