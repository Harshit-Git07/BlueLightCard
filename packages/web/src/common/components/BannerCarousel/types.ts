import { CampaignCardProps } from '../CampaignCard/types';

export type BannerCarouselPropers = {
  banners: CampaignCardProps[];
};
