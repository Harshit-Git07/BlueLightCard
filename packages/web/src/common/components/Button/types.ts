import { ThemeVariant } from '@/types/theme';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { MouseEventHandler, PropsWithChildren } from 'react';

export type ButtonProps = PropsWithChildren & {
  type?: 'button' | 'submit' | 'link';
  disabled?: boolean;
  slim?: boolean;
  className?: string;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  onClick?: MouseEventHandler;
};
