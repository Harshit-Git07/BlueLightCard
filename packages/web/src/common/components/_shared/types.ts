import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ChangeEventHandler, ForwardedRef, MouseEventHandler, PropsWithChildren } from 'react';

export type InputFieldWrapperProps = PropsWithChildren & {
  icon?: IconDefinition;
  iconRight?: IconDefinition;
  showRightIcon?: boolean;
  showErrorState?: boolean;
  select?: boolean;
  showSuccessState?: boolean;
  onRightIconClick?: MouseEventHandler;
};

export interface InputFieldSharedProps<E, V = string> {
  name?: string;
  error?: boolean;
  success?: boolean;
  value?: V;
  required?: boolean;
  _ref?: ForwardedRef<unknown>;
  onChange?: ChangeEventHandler<E>;
}
