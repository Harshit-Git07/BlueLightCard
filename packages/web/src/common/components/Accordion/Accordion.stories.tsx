import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import Accordion from './Accordion';
import { AccordionProps } from './types';

export default {
  title: 'Component System/Accordion',
  component: Accordion,
} as Meta;

const Template: StoryFn<AccordionProps> = (args) => <Accordion {...args} />;

export const AccordionBox = Template.bind({});
AccordionBox.args = {
  title: 'Terms and Conditions',
  content: 'These are the terms and conditions for the above offer.',
};

export const AccordionBoxMarkdownBold = Template.bind({});
AccordionBoxMarkdownBold.args = {
  title: 'Terms and Conditions',
  content: 'These are **bold** terms and conditions for the above offer.',
};
export const AccordionBoxMarkdownItalic = Template.bind({});
AccordionBoxMarkdownItalic.args = {
  title: 'Terms and Conditions',
  content: 'These are *italic* terms and conditions for the above offer.',
};
export const AccordionBoxMarkdownStrong = Template.bind({});
AccordionBoxMarkdownStrong.args = {
  title: 'Terms and Conditions',
  content: 'These are __strong__ terms and conditions for the above offer.',
};
export const AccordionBoxMarkdownNewLine = Template.bind({});
AccordionBoxMarkdownNewLine.args = {
  title: 'Terms and Conditions',
  content: 'These are \n terms and conditions \n on a new line \n for the above offer.',
};

export const AccordionBoxMarkdownStrikeThrough = Template.bind({});
AccordionBoxMarkdownStrikeThrough.args = {
  title: 'Terms and Conditions',
  content: '~~These are terms and conditions on a new line for the above offer.~~',
};
