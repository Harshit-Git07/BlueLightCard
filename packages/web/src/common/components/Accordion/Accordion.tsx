import React, { FC, useState } from 'react';
import { AccordionProps } from '@/components/Accordion/types';
import Markdown from 'markdown-to-jsx';

const Accordion: FC<AccordionProps> = ({ title, content }) => {
  const [active, setActive] = useState(false);

  const handleToggle = () => {
    setActive(!active);
  };
  return (
    <div className="mb-8 w-full p-4 sm:p-8 lg:px-6 xl:px-8">
      <button className={`flex w-full text-left`} onClick={() => handleToggle()}>
        <div className="w-full">
          <h4 className="mt-1.5 font-['MuseoSans'] text-lg font-semibold text-shade-greyscale-grey-900">
            {title}
          </h4>
        </div>
        <div className="mr-5 flex h-10 w-full max-w-[40px] items-center justify-center">
          <span className="icon inline-flex h-12 w-full max-w-[32px] items-center justify-center text-3xl font-semibold text-[#001B80]">
            {active ? '-' : '+'}
          </span>
        </div>
      </button>

      <div className={`duration-200 ease-in-out ${active ? 'block' : 'hidden'}`}>
        <Markdown className="font-['MuseoSans'] py-3 text-shade-greyscale-grey-900 whitespace-pre-wrap">
          {content}
        </Markdown>
      </div>
    </div>
  );
};

export default Accordion;
