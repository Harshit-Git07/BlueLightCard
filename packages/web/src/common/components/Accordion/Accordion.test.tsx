import Accordion from '@/components/Accordion/Accordion';
import { AccordionProps } from '@/components/Accordion/types';
import renderer from 'react-test-renderer';

describe('Accordion component', () => {
  let props: AccordionProps;

  it('should render an accordion', () => {
    const props = {
      title: 'Terms and Conditions',
      content: 'This is an accordion message.',
    };

    const component = renderer.create(<Accordion {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an accordion with bold text', () => {
    const props = {
      title: 'Terms and Conditions',
      content: 'This is an **accordion** message.',
    };

    const component = renderer.create(<Accordion {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an accordion with italic text', () => {
    const props = {
      title: 'Terms and Conditions',
      content: 'This is an *accordion* message.',
    };

    const component = renderer.create(<Accordion {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an accordion with strong text', () => {
    const props = {
      title: 'Terms and Conditions',
      content: 'This is an __accordion__ message.',
    };

    const component = renderer.create(<Accordion {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an accordion with new lines text', () => {
    const props = {
      title: 'Terms and Conditions',
      content: 'This is an \naccordion\n message.',
    };

    const component = renderer.create(<Accordion {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an accordion with strike through text', () => {
    const props = {
      title: 'Terms and Conditions',
      content: '~~This is an accordion message.~~',
    };

    const component = renderer.create(<Accordion {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
