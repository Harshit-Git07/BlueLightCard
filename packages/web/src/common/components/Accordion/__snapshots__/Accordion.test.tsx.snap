// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component should render an accordion 1`] = `
<div
  className="mb-8 w-full p-4 sm:p-8 lg:px-6 xl:px-8"
>
  <button
    className="flex w-full text-left"
    onClick={[Function]}
  >
    <div
      className="w-full"
    >
      <h4
        className="mt-1.5 font-['MuseoSans'] text-lg font-semibold text-shade-greyscale-grey-900"
      >
        Terms and Conditions
      </h4>
    </div>
    <div
      className="mr-5 flex h-10 w-full max-w-[40px] items-center justify-center"
    >
      <span
        className="icon inline-flex h-12 w-full max-w-[32px] items-center justify-center text-3xl font-semibold text-[#001B80]"
      >
        +
      </span>
    </div>
  </button>
  <div
    className="duration-200 ease-in-out hidden"
  >
    <span
      className="font-['MuseoSans'] py-3 text-shade-greyscale-grey-900 whitespace-pre-wrap"
    >
      This is an accordion message.
    </span>
  </div>
</div>
`;

exports[`Accordion component should render an accordion with bold text 1`] = `
<div
  className="mb-8 w-full p-4 sm:p-8 lg:px-6 xl:px-8"
>
  <button
    className="flex w-full text-left"
    onClick={[Function]}
  >
    <div
      className="w-full"
    >
      <h4
        className="mt-1.5 font-['MuseoSans'] text-lg font-semibold text-shade-greyscale-grey-900"
      >
        Terms and Conditions
      </h4>
    </div>
    <div
      className="mr-5 flex h-10 w-full max-w-[40px] items-center justify-center"
    >
      <span
        className="icon inline-flex h-12 w-full max-w-[32px] items-center justify-center text-3xl font-semibold text-[#001B80]"
      >
        +
      </span>
    </div>
  </button>
  <div
    className="duration-200 ease-in-out hidden"
  >
    <span
      className="font-['MuseoSans'] py-3 text-shade-greyscale-grey-900 whitespace-pre-wrap"
    >
      This is an 
      <strong>
        accordion
      </strong>
       message.
    </span>
  </div>
</div>
`;

exports[`Accordion component should render an accordion with italic text 1`] = `
<div
  className="mb-8 w-full p-4 sm:p-8 lg:px-6 xl:px-8"
>
  <button
    className="flex w-full text-left"
    onClick={[Function]}
  >
    <div
      className="w-full"
    >
      <h4
        className="mt-1.5 font-['MuseoSans'] text-lg font-semibold text-shade-greyscale-grey-900"
      >
        Terms and Conditions
      </h4>
    </div>
    <div
      className="mr-5 flex h-10 w-full max-w-[40px] items-center justify-center"
    >
      <span
        className="icon inline-flex h-12 w-full max-w-[32px] items-center justify-center text-3xl font-semibold text-[#001B80]"
      >
        +
      </span>
    </div>
  </button>
  <div
    className="duration-200 ease-in-out hidden"
  >
    <span
      className="font-['MuseoSans'] py-3 text-shade-greyscale-grey-900 whitespace-pre-wrap"
    >
      This is an 
      <em>
        accordion
      </em>
       message.
    </span>
  </div>
</div>
`;

exports[`Accordion component should render an accordion with new lines text 1`] = `
<div
  className="mb-8 w-full p-4 sm:p-8 lg:px-6 xl:px-8"
>
  <button
    className="flex w-full text-left"
    onClick={[Function]}
  >
    <div
      className="w-full"
    >
      <h4
        className="mt-1.5 font-['MuseoSans'] text-lg font-semibold text-shade-greyscale-grey-900"
      >
        Terms and Conditions
      </h4>
    </div>
    <div
      className="mr-5 flex h-10 w-full max-w-[40px] items-center justify-center"
    >
      <span
        className="icon inline-flex h-12 w-full max-w-[32px] items-center justify-center text-3xl font-semibold text-[#001B80]"
      >
        +
      </span>
    </div>
  </button>
  <div
    className="duration-200 ease-in-out hidden"
  >
    <p
      className="font-['MuseoSans'] py-3 text-shade-greyscale-grey-900 whitespace-pre-wrap"
    >
      This is an 
accordion
 message.
    </p>
  </div>
</div>
`;

exports[`Accordion component should render an accordion with strike through text 1`] = `
<div
  className="mb-8 w-full p-4 sm:p-8 lg:px-6 xl:px-8"
>
  <button
    className="flex w-full text-left"
    onClick={[Function]}
  >
    <div
      className="w-full"
    >
      <h4
        className="mt-1.5 font-['MuseoSans'] text-lg font-semibold text-shade-greyscale-grey-900"
      >
        Terms and Conditions
      </h4>
    </div>
    <div
      className="mr-5 flex h-10 w-full max-w-[40px] items-center justify-center"
    >
      <span
        className="icon inline-flex h-12 w-full max-w-[32px] items-center justify-center text-3xl font-semibold text-[#001B80]"
      >
        +
      </span>
    </div>
  </button>
  <div
    className="duration-200 ease-in-out hidden"
  >
    <del
      className="font-['MuseoSans'] py-3 text-shade-greyscale-grey-900 whitespace-pre-wrap"
    >
      This is an accordion message.
    </del>
  </div>
</div>
`;

exports[`Accordion component should render an accordion with strong text 1`] = `
<div
  className="mb-8 w-full p-4 sm:p-8 lg:px-6 xl:px-8"
>
  <button
    className="flex w-full text-left"
    onClick={[Function]}
  >
    <div
      className="w-full"
    >
      <h4
        className="mt-1.5 font-['MuseoSans'] text-lg font-semibold text-shade-greyscale-grey-900"
      >
        Terms and Conditions
      </h4>
    </div>
    <div
      className="mr-5 flex h-10 w-full max-w-[40px] items-center justify-center"
    >
      <span
        className="icon inline-flex h-12 w-full max-w-[32px] items-center justify-center text-3xl font-semibold text-[#001B80]"
      >
        +
      </span>
    </div>
  </button>
  <div
    className="duration-200 ease-in-out hidden"
  >
    <span
      className="font-['MuseoSans'] py-3 text-shade-greyscale-grey-900 whitespace-pre-wrap"
    >
      This is an 
      <strong>
        accordion
      </strong>
       message.
    </span>
  </div>
</div>
`;
